docs/configure-archive-logs.md

```diff
@@ -59,3 +60,14 @@ spec:
     archiveLocation:
       archiveLogs: true
 ```
+
+## Why doesn't Argo Workflows recommend using this feature?
```

Comment:
maybe "Why doesn't Argo Workflows recommend relying on this feature?" or "Why doesn't Argo Workflow recommend using this feature in production?"

Reply:
or other words that don't suggest that somehow the software is recommending against using a feature of itself. Maybe "the Argo Maintainers" instead of "Argo Workflows"?

Reply:
Changed after [below comments](https://github.com/argoproj/argo-workflows/pull/12597#discussion_r1473454054)